tests/andromede/test_electrolyzer_n_inputs.py

```diff
@@ -0,0 +1,301 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+from andromede.libs.standard import DEMAND_MODEL, GENERATOR_MODEL, NODE_BALANCE_MODEL
+from andromede.libs.standard_sc import (
+    CONVERTOR_MODEL,
+    CONVERTOR_MODEL_MOD,
+    DECOMPOSE_1_FLOW_INTO_2_FLOW,
+    TWO_INPUTS_CONVERTOR_MODEL,
+)
+from andromede.simulation import TimeBlock, build_problem
+from andromede.study import (
+    ConstantData,
+    DataBase,
+    Network,
+    Node,
+    PortRef,
+    create_component,
+)
+
+
+def test_electrolyzer_n_inputs_1():
+    """
+    Test with an electrolyzer for each inputs
+    """
+    elec_node_1 = Node(model=NODE_BALANCE_MODEL, id="e1")
+    electric_prod_1 = create_component(model=GENERATOR_MODEL, id="ep1")
+    electrolyzer1 = create_component(model=CONVERTOR_MODEL, id="ez1")
+
+    elec_node_2 = Node(model=NODE_BALANCE_MODEL, id="e2")
+    electric_prod_2 = create_component(model=GENERATOR_MODEL, id="ep2")
+    electrolyzer2 = create_component(model=CONVERTOR_MODEL, id="ez2")
+
+    gaz_node = Node(model=NODE_BALANCE_MODEL, id="g")
+    gaz_prod = create_component(model=GENERATOR_MODEL, id="gp")
+    gaz_demand = create_component(model=DEMAND_MODEL, id="gd")
+
+    database = DataBase()
+
+    database.add_data("ep1", "p_max", ConstantData(100))
+    database.add_data("ep1", "cost", ConstantData(30))
+    database.add_data("ez1", "alpha", ConstantData(0.7))
+
+    database.add_data("ep2", "p_max", ConstantData(100))
+    database.add_data("ep2", "cost", ConstantData(30))
+    database.add_data("ez2", "alpha", ConstantData(0.7))
+
+    database.add_data("gd", "demand", ConstantData(70))
```

Comment:
It could be more interesting to have a higher demand that needs production from both electrolyzers